service/controller/clusterapi/v29/resource/clusterazs/create.go

```diff
@@ -31,12 +30,22 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 		return microerror.Mask(err)
 	}
 
-	var machineDeployments []clusterv1alpha1.MachineDeployment
+	// We need to cancel the resource early in case the ipam resource did not yet
+	// allocate a subnet for the tenant cluster.
+	if key.StatusClusterNetworkCIDR(cr) == "" {
+		r.logger.LogCtx(ctx, "level", "debug", "message", "cannot collect subnets for availability zones")
+		r.logger.LogCtx(ctx, "level", "debug", "message", "cluster network not yet allocated")
+		r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource")
```

Comment:
Moving that up. No need to do a lot of work when we cannot make use of the result anyway.